src/task.c

```diff
@@ -507,6 +518,17 @@ JL_DLLEXPORT void jl_switch(void)
         jl_error("cannot switch to task running on another thread");
     }
 
+    if (jl_task_switch_hooks) {
```

Comment:
Do we need a lock (or some atomics) here? If so, wouldn't it increase the overhead of the task switch even if there are no hooks?